src/schema/posts.ts

```diff
@@ -859,6 +925,80 @@ export const resolvers: IResolvers<any, Context> = {
 
       return { _: true };
     },
+    editPost: async (
+      source,
+      args: EditPostArgs,
+      ctx: Context,
+    ): Promise<GQLEmptyResponse> => {
+      const { id, image } = args;
+      const { con, userId } = ctx;
+      const title = args.title?.trim() ?? '';
+      const content = args.content?.trim() ?? '';
+
+      await con.transaction(async (manager) => {
+        const repo = manager.getRepository(Post);
+        const post = (await repo.findOneByOrFail({ id })) as
+          | WelcomePost
+          | FreeformPost;
+
+        if (post.authorId !== userId) {
+          const permission =
+            post.type === PostType.Welcome
+              ? SourcePermissions.WelcomePostEdit
+              : SourcePermissions.Post;
+          await ensureSourcePermissions(ctx, post.sourceId, permission);
+        }
+
+        if (!editablePostTypes.includes(post.type)) {
+          throw new ForbiddenError(
+            'Editing post outside of type welcome_post and freeform is not allowed',
+          );
+        }
+
+        if (title.length > MAX_TITLE_LENGTH) {
+          throw new ValidationError(
+            'Title has a maximum length of 80 characters',
+          );
+        }
+
+        if (content.length > MAX_CONTENT_LENGTH) {
+          throw new ValidationError(
+            'Content has a maximum length of 80 characters',
+          );
+        }
+
+        const updated: Partial<EditablePost> = {};
+
+        if (post.type === PostType.Freeform) {
+          updated.title = title;
+
+          if (image && process.env.CLOUDINARY_URL) {
```

Comment:
Not blocking but just checking on how do we validate required `process.env` in general? Should we just throw here or at least log that `CLOUDINARY_URL` is missing. Looks like something that could silently happen and it would require quite some times to figure out in production.

Reply:
We don't have a staging environment to test this out so we manually test this on prod. This is an existing implementation for the user update profile, etc., so if one does not work, it won't across the board.

Reply:
Don't mind it for this PR, your implementation is fine for sure.. just something that caught my eye, maybe a backlog candidate.

Reply:
Definitely, also that we don't have any setup for at least mocking this in our test cases.